include/zephyr/net/dns_resolve.h

```diff
@@ -44,7 +44,7 @@ enum dns_query_type {
 
 /** Max size of the resolved name. */
 #ifndef DNS_MAX_NAME_SIZE
-#define DNS_MAX_NAME_SIZE 20
+#define DNS_MAX_NAME_SIZE 32
 #endif
```

Comment:
Instead of hardcoding the value here, we could allow user to change the value via a Kconfig, so could you create a Kconfig option for this and set the default value to the present value (20).